src/mlpack/methods/ann/gan_impl.hpp

```diff
@@ -127,8 +129,9 @@ double GAN<Model, InitializationRuleType, Noise>::Evaluate(
   if (!reset)
     Reset();
 
-  currentInput = this->predictors.unsafe_col(i);
-  currentTarget = this->responses.unsafe_col(i);
+  // Alias here
+  currentInput = this->predictors.cols(i, i + batchSize - 1);
+  currentTarget = this->responses.cols(i, i + batchSize - 1);
```

Comment:
@zoq Is there any better way of aliasing the columns here? If not I will remove the above comment.

Reply:
hm, we could use ``memptr()`` (advanced constructor), what do you think?